src/api/documents/document-ref.ts

```diff
@@ -130,18 +111,39 @@ export class DocumentRef<T extends { [key: string]: any }> {
     });
   }
 
+  /**
+   * 
+   */
+  public toWire(): Key {
+    const key = new Key();
+    key.setCollection(this.parent.toWire());
+    key.setId(this.id);
+
+    return key;
+  }
+
+  private depth(depth = 0): number {
+    const parent = this.parent.parent;
+
+    if (parent) {
+      depth = parent.depth(depth + 1);
+    }
+
+    return depth;
+  }
```

Comment:
Same comment about this depth calc as the one for collection groups

Reply:
done